error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -556,6 +557,152 @@ Mono<T> after(Mono<T> mono) {
     }
   }
 
+  /**
+   * Don't unnecessarily transform a {@link Flux#using(Callable, Function)} to a mono, instead use
+   * the equivalent API provided by {@link Mono}.
+   */
+  static final class MonoUsing<
+      D extends AutoCloseable, T, P extends Publisher<? extends T>, M extends Mono<? extends T>> {
+    @BeforeTemplate
+    Mono<T> before(
+        Callable<D> resourceSupplier,
+        @Matches(IsFunctionReturningMono.class) Function<D, P> sourceSupplier) {
```

Comment:
@Stephan202 had the idea to see if we can simply define P here by saying `P extends Mono<? extends T>` but I couldn't get that to work. I think as then the `Publisher` overload is not being used and Refaster cannot match it anymore.